### ISSUE TITLE
get_lp() uncallable

### ISSUE THREAD

**Issue by bgoodri** [stan-dev/stan#1500]
Parsing
```
functions {
  void foo_lp() {
    print("lp__ = ", get_lp());
  }
}
model {}
```
yields
```
SYNTAX ERROR, MESSAGE(S) FROM PARSER:

lp suffixed functions only allowed in transformed parameter, function argument, or model in block=void function argument '_lp' suffixed
```
which is not really English, not really necessary for the `get_lp()` function, and doesn't really make sense because a user-defined function that touches lp__ could be called in transformed parameters or model.

**Comment by syclik** [comment]
@bgoodri, can you tag this with a milestone?

**Comment by bgoodri** [comment]
OK. Maybe we should just make an intelligible error message for 2.7.0, re-allow get_lp() for 2.7.1, and fix the logic for 2.8.0?

**Comment by bob-carpenter** [comment]
I fixed both the logic of where it can be called and the
error message to make it more intelligible.  It looks like
a line of error message was somehow just cut out --- there was
a blank line where there shouldn't have been that would've
filled in the garbled text.  

- Bob
